### PR TITLE
[copp tables] validate copp tables during in/cross-branch upgrades

### LINKED ISSUE #8085: [testgap] validate preserving copp tables work during in/cross-branch upgrades
**Description**
This is regarding https://github.com/sonic-net/SONiC/pull/1107

The design added a change where COPP table is preserved for MLNX platforms.

With this new change a test gap has surfaced to ensure:
Fast and warm upgrade is not broken - in/cross-branch.
Cross branch upgrade has no issues (eg., 201911 -> 202205): copp table entries after upgrade are as expected. This is critical to ensure that there are no crashes, the entries from 201911 are deleted, entries from 202205 are as expected.
The keys and corresponding values are as expected in appl_db.
All traps from base OS are still functional on the target OS.

**Steps to reproduce the issue:**
1.
2.
3.

**Describe the results you received:**

**Describe the results you expected:**

**Additional information you deem important:**

    **Output of `show version`:**

    ```
    (paste your output here)
    ```

    **Attach debug file `sudo generate_dump`:**

    ```
    (paste your output here)
    ```

### PR DESCRIPTION
**Pull request by AntonHryshchuk**
### Description of PR

Summary: Update the test test_upgrade_path.py to support validation of the copp tables after in/cross-branch upgrades.
Added validation after upgrade, to compare copp tables with copp_cfg.json file.
Fixes #8085 

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [X] 202205

#### What is the motivation for this PR?
Add validation of copp tables after improvement of fast-reboot

#### How did you do it?
Added validation of copp tables after upgrade 

#### How did you verify/test it?
executed test _test_upgrade_path_ with _base_image=**201911**, target_image=**202205**, upgrade_type=**fast**_

### CODE REVIEW

**Review by arfeigin — APPROVED** [review]
LGTM

**Comment by arfeigin** [comment]
Hi @vaibhavhd , please review the test changes.

**Comment by liat-grozovik** [comment]
@vaibhavhd would you like to review and provide your feedback?

File: tests/upgrade_path/upgrade_helpers.py [hunk 1]
```diff
@@ -117,3 +119,51 @@ def check_reboot_cause(duthost, expected_cause):
     reboot_cause = get_reboot_cause(duthost)
     logging.info("Checking cause from dut {} to expected {}".format(reboot_cause, expected_cause))
     return reboot_cause == expected_cause
+
+
+def check_copp_config(duthost):
+    logging.info("Comparing CoPP configuration from copp_cfg.json to COPP_TABLE")
+    copp_tables = json.loads(duthost.shell("sonic-db-dump -n APPL_DB -k COPP_TABLE* -y")["stdout"])
+    copp_cfg = json.loads(duthost.shell("cat /etc/sonic/copp_cfg.json")["stdout"])
```
Comment (vaibhavhd) [inline tests/upgrade_path/upgrade_helpers.py:127]: Should there be a check for existence of config prior to this step?

File: tests/upgrade_path/upgrade_helpers.py [hunk 2]
```diff
@@ -117,3 +119,51 @@ def check_reboot_cause(duthost, expected_cause):
...
+    feature_status = duthost.shell("show feature status")["stdout"]
+    copp_tables_formatted = get_copp_table_formatted_dict(copp_tables)
+    copp_cfg_formatted = get_copp_cfg_formatted_dict(copp_cfg, feature_status)
+    pytest_assert(copp_tables_formatted == copp_cfg_formatted,
+                  "There is a difference between CoPP config and CoPP tables. CoPP config: {}\nCoPP tables:"
+                  " {}".format(copp_tables_formatted, copp_cfg_formatted))
```
Comment (vaibhavhd) [inline tests/upgrade_path/upgrade_helpers.py:133]: The print sequence is reverted.

**Review by vaibhavhd — APPROVED** [review]
Non blocking comments added

**Pull request merged**